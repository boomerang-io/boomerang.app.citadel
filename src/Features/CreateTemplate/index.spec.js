import React from "react";
import Inputs from "../CreateTemplate";
import { fireEvent } from "@testing-library/react";

const mockfn = jest.fn();

const initialState = {
  workflow: {
    data: {
      rules: [
        {
          defaultValue: "pandas",
          description: "Tim property",
          key: "tim.property",
          label: "Tim Property",
          required: true,
          type: "select",
          validValues: ["pandas", "dogs"]
        }
      ]
    }
  }
};

const props = {
  loading: false,
  updateInputs: mockfn,
  workflowActions: { deleteWorkflowInput: mockfn }
};

beforeEach(() => {
  document.body.setAttribute("id", "app");
});

describe("Inputs --- Snapshot Test", () => {
  it("Capturing Snapshot of Inputs", () => {
    const { baseElement } = rtlReduxRender(<Inputs {...props} />, { initialState });
    expect(baseElement).toMatchSnapshot();
  });
});

