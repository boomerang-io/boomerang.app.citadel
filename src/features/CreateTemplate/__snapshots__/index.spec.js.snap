// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inputs --- Snapshot Test Capturing Snapshot of Inputs 1`] = `
<body
  id="app"
>
  <div>
    <form
      action="#"
    >
      <div
        class="container"
      >
        <div
          class="content"
        >
          <div
            class="info"
          >
            <h1
              class="title"
            >
              Create Template
            </h1>
          </div>
          <section
            class="buttons"
          >
            <button
              class="button bx--btn bx--btn--field bx--btn--secondary"
              tabindex="0"
              type="button"
            >
              Cancel
            </button>
            <button
              class="button bx--btn bx--btn--field bx--btn--primary"
              data-testid="policy-header-affirmative-action"
              tabindex="0"
              type="submit"
            >
              Create
              <svg
                aria-hidden="true"
                aria-label="Create"
                class="bx--btn__icon"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                role="img"
                style="will-change: transform;"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 7V3H7v4H3v2h4v4h2V9h4V7z"
                />
              </svg>
            </button>
          </section>
        </div>
      </div>
      <section
        class="container"
      >
        <div
          class="bx--tabs"
          role="navigation"
        >
          <div
            aria-label="listbox"
            class="bx--tabs-trigger"
            role="listbox"
            tabindex="0"
          >
            <a
              class="bx--tabs-trigger-text"
              href="#"
              tabindex="-1"
            >
              About
            </a>
            <svg
              aria-hidden="true"
              focusable="false"
              height="6"
              preserveAspectRatio="xMidYMid meet"
              style="will-change: transform;"
              viewBox="0 0 10 6"
              width="10"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 6L0 1 .7.3 5 4.6 9.3.3l.7.7z"
              />
              <title>
                show menu options
              </title>
            </svg>
          </div>
          <ul
            class="bx--tabs__nav bx--tabs__nav--hidden"
            role="tablist"
          >
            <li
              class="bx--tabs__nav-item bx--tabs__nav-item--selected"
              role="presentation"
              tabindex="-1"
            >
              <a
                aria-selected="true"
                class="bx--tabs__nav-link"
                href="#"
                role="tab"
                tabindex="0"
              >
                About
              </a>
            </li>
            <li
              class="bx--tabs__nav-item"
              role="presentation"
              tabindex="-1"
            >
              <a
                aria-selected="false"
                class="bx--tabs__nav-link"
                href="#"
                role="tab"
                tabindex="-1"
              >
                Rules
              </a>
            </li>
            <li
              class="bx--tabs__nav-item"
              role="presentation"
              tabindex="-1"
            >
              <a
                aria-selected="false"
                class="bx--tabs__nav-link"
                href="#"
                role="tab"
                tabindex="-1"
              >
                OPA Rego
              </a>
            </li>
          </ul>
        </div>
        <div
          aria-hidden="false"
        >
          <div
            class="generalContainer"
          >
            <section
              class="generalSection"
            >
              <h1
                class="sectionTitle"
              >
                General
              </h1>
              <div
                class="bx--bmrg-text-input"
              >
                <div
                  class="bx--form-item bx--text-input-wrapper"
                >
                  <label
                    class="bx--label"
                    for="name"
                  >
                    <div
                      class="bx--bmrg-text-input__label"
                    >
                      <div>
                        Name
                      </div>
                    </div>
                  </label>
                  <div
                    class="bx--text-input__field-wrapper"
                  >
                    <input
                      class="bx--text-input bx--text__input"
                      id="name"
                      name="name"
                      placeholder="Name"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="bx--bmrg-text-input"
              >
                <div
                  class="bx--form-item bx--text-input-wrapper"
                >
                  <label
                    class="bx--label"
                    for="key"
                  >
                    <div
                      class="bx--bmrg-text-input__label"
                    >
                      <div>
                        Key
                      </div>
                    </div>
                  </label>
                  <div
                    class="bx--form__helper-text"
                  >
                    Must match the OPA Rego package declaration
                  </div>
                  <div
                    class="bx--text-input__field-wrapper"
                  >
                    <input
                      class="bx--text-input bx--text__input"
                      id="key"
                      name="key"
                      placeholder="Key"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="bx--bmrg-text-area"
              >
                <div
                  class="bx--form-item"
                >
                  <label
                    class="bx--label"
                    for="description"
                  >
                    <div
                      style="display: flex;"
                    >
                      <div>
                        Description (optional)
                      </div>
                    </div>
                  </label>
                  <div
                    class="bx--text-area__wrapper"
                  >
                    <textarea
                      class="bx--text-area"
                      cols="50"
                      id="description"
                      name="description"
                      placeholder="Description"
                      rows="4"
                    />
                  </div>
                </div>
              </div>
              <div
                class="bx--bmrg-text-input"
              >
                <div
                  class="bx--form-item bx--text-input-wrapper"
                >
                  <label
                    class="bx--label"
                    for="order"
                  >
                    <div
                      class="bx--bmrg-text-input__label"
                    >
                      <div>
                        Order (optional)
                      </div>
                    </div>
                  </label>
                  <div
                    class="bx--form__helper-text"
                  >
                    Specify presentation order in Create Policy feature
                  </div>
                  <div
                    class="bx--text-input__field-wrapper"
                  >
                    <input
                      class="bx--text-input bx--text__input"
                      id="order"
                      name="order"
                      placeholder="0"
                      type="number"
                      value="0"
                    />
                  </div>
                </div>
              </div>
            </section>
            <section
              class="generalSection"
            >
              <h1
                class="sectionTitle"
              >
                Validation
              </h1>
              <div
                class="bx--bmrg-creatable"
              >
                <div
                  class="bx--bmrg-creatable__input"
                >
                  <div
                    class="bx--form-item bx--text-input-wrapper"
                  >
                    <label
                      class="bx--label"
                      for="labels"
                    >
                      <div
                        style="display: flex;"
                      >
                        <div>
                          Labels (optional)
                        </div>
                      </div>
                    </label>
                    <div
                      class="bx--form__helper-text"
                    >
                      Metadata to pass information into the pre-integrated repositories
                    </div>
                    <div
                      class="bx--text-input__field-wrapper"
                    >
                      <input
                        class="bx--text-input bx--text__input"
                        id="labels"
                        maxlength="64"
                        placeholder="Create labels"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <button
                    class="bx--bmrg-creatable__button bx--btn bx--btn--field bx--btn--primary bx--btn--disabled"
                    disabled=""
                    tabindex="0"
                    type="button"
                  >
                    Add
                    <svg
                      aria-hidden="true"
                      aria-label="Add"
                      class="bx--btn__icon"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      viewBox="0 0 32 32"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <polygon
                        points="17,15 17,8 15,8 15,15 8,15 8,17 15,17 15,24 17,24 17,17 24,17 24,15"
                      />
                    </svg>
                  </button>
                </div>
                <div
                  class="bx--bmrg-creatable__tags"
                />
              </div>
            </section>
          </div>
        </div>
        <div
          aria-hidden="true"
          hidden=""
        >
          <div
            class="propertyList"
          >
            <button
              class="createPropertyCard"
              data-testid="create-new-workflow-input-button"
              type="button"
            >
              <div
                class="createContainer"
              >
                <svg
                  aria-label="Add"
                  class="createIcon"
                  focusable="false"
                  height="32"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  style="will-change: transform;"
                  viewBox="0 0 32 32"
                  width="32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M17 15V7h-2v8H7v2h8v8h2v-8h8v-2h-8z"
                  />
                </svg>
                <p
                  class="createText"
                >
                  Create a new rule
                </p>
              </div>
            </button>
          </div>
        </div>
        <div
          aria-hidden="true"
          hidden=""
        >
          <section
            class="opaPolicyContainer"
          >
            <div
              style="max-width: 80rem; max-height: 50rem; height: 100%; width: 100%; margin: auto; overflow-x: visible; display: flex; flex-direction: column; justify-content: flex-start;"
            >
              <div
                class="bx--toolbar b-task-text-area"
              >
                <button
                  class="b-task-text-area__button bx--btn bx--btn--sm bx--btn--ghost bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--bottom bx--tooltip--align-end"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="bx--assistive-text"
                  >
                    Undo
                  </span>
                  <svg
                    aria-hidden="true"
                    aria-label="Undo"
                    class="bx--btn__icon"
                    focusable="false"
                    height="20"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    style="will-change: transform;"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M15 9.3c1.5 0 2.8 1.3 2.8 2.7 0 1.5-1.3 2.7-2.8 2.7h-4V16h4c2.3 0 4-1.7 4-4 0-2.2-1.8-4-4-4H3.4l2.5-2.5-.9-.9-4 4 4 4 .9-.9-2.5-2.4H15z"
                    />
                  </svg>
                </button>
                <button
                  class="b-task-text-area__button bx--btn bx--btn--sm bx--btn--ghost bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--bottom bx--tooltip--align-center"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="bx--assistive-text"
                  >
                    Redo
                  </span>
                  <svg
                    aria-hidden="true"
                    aria-label="Redo"
                    class="bx--btn__icon"
                    focusable="false"
                    height="20"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    style="will-change: transform;"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M5 9.3c-1.5 0-2.8 1.3-2.8 2.7 0 1.5 1.3 2.7 2.8 2.7h4V16H5c-2.3 0-4-1.7-4-4 0-2.2 1.8-4 4-4h11.6l-2.5-2.5.9-.9 4 4-4 4-.9-.9 2.5-2.5H5z"
                    />
                  </svg>
                </button>
                <button
                  class="b-task-text-area__button bx--btn bx--btn--sm bx--btn--ghost bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--bottom bx--tooltip--align-center"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="bx--assistive-text"
                  >
                    Copy
                  </span>
                  <svg
                    aria-hidden="true"
                    aria-label="Copy"
                    class="bx--btn__icon"
                    focusable="false"
                    height="20"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    style="will-change: transform;"
                    viewBox="0 0 32 32"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M28 10v18H10V10h18m0-2H10a2 2 0 0 0-2 2v18a2 2 0 0 0 2 2h18a2 2 0 0 0 2-2V10a2 2 0 0 0-2-2z"
                    />
                    <path
                      d="M4 18H2V4a2 2 0 0 1 2-2h14v2H4z"
                    />
                  </svg>
                </button>
                <button
                  class="b-task-text-area__button bx--btn bx--btn--sm bx--btn--ghost bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--bottom bx--tooltip--align-center"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="bx--assistive-text"
                  >
                    Cut
                  </span>
                  <svg
                    aria-hidden="true"
                    aria-label="Cut"
                    class="bx--btn__icon"
                    focusable="false"
                    height="20"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    style="will-change: transform;"
                    viewBox="0 0 32 32"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M26.5 19.63L20.24 16l6.26-3.63a5 5 0 0 0-1.21-9.2A5.19 5.19 0 0 0 24 3a5 5 0 0 0-4.33 7.53 5 5 0 0 0 2.39 2.1l-3.82 2.21L4 6.6 3 8.34 16.24 16 3 23.68l1 1.74 14.24-8.26 3.82 2.21a5 5 0 0 0-2.39 2.1A5 5 0 0 0 24 29a5.19 5.19 0 0 0 1.29-.17 5 5 0 0 0 1.21-9.2zm-5.1-10.1a3 3 0 0 1 1.1-4.12 3 3 0 0 1 4.1 1.11 3 3 0 0 1-1.1 4.11 3 3 0 0 1-4.1-1.1zm5.2 16a3 3 0 0 1-4.1 1.11 3 3 0 0 1-1.1-4.12 3 3 0 0 1 4.1-1.1 3 3 0 0 1 1.1 4.06z"
                    />
                  </svg>
                </button>
                <button
                  class="b-task-text-area__button bx--btn bx--btn--sm bx--btn--ghost bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--bottom bx--tooltip--align-center"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="bx--assistive-text"
                  >
                    Paste
                  </span>
                  <svg
                    aria-hidden="true"
                    aria-label="Paste"
                    class="bx--btn__icon"
                    focusable="false"
                    height="20"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    style="will-change: transform;"
                    viewBox="0 0 32 32"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M26 20h-8.17l2.58-2.59L19 16l-5 5 5 5 1.41-1.41L17.83 22H26v8h2v-8a2 2 0 0 0-2-2z"
                    />
                    <path
                      d="M23.71 9.29l-7-7A1 1 0 0 0 16 2H6a2 2 0 0 0-2 2v24a2 2 0 0 0 2 2h8v-2H6V4h8v6a2 2 0 0 0 2 2h6v2h2v-4a1 1 0 0 0-.29-.71zM16 4.41L21.59 10H16z"
                    />
                  </svg>
                </button>
                <div
                  class="bx--search bx--search--sm"
                >
                  <svg
                    aria-hidden="true"
                    class="bx--search-magnifier"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    style="will-change: transform;"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M15 14.3L10.7 10c1.9-2.3 1.6-5.8-.7-7.7S4.2.7 2.3 3 .7 8.8 3 10.7c2 1.7 5 1.7 7 0l4.3 4.3.7-.7zM2 6.5C2 4 4 2 6.5 2S11 4 11 6.5 9 11 6.5 11 2 9 2 6.5z"
                    />
                  </svg>
                  <label
                    class="bx--label"
                    for="search"
                  >
                    Search
                  </label>
                  <input
                    class="bx--search-input"
                    id="search"
                    placeholder="Search"
                    type="text"
                  />
                  <button
                    aria-label=""
                    class="bx--search-close bx--search-close--hidden"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      style="will-change: transform;"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 4.7l-.7-.7L8 7.3 4.7 4l-.7.7L7.3 8 4 11.3l.7.7L8 8.7l3.3 3.3.7-.7L8.7 8z"
                      />
                    </svg>
                  </button>
                </div>
                <button
                  class="b-task-text-area__button bx--btn bx--btn--sm bx--btn--ghost bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--bottom bx--tooltip--align-center"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="bx--assistive-text"
                  >
                    Find previous
                  </span>
                  <svg
                    aria-hidden="true"
                    aria-label="Find previous"
                    class="bx--btn__icon"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    style="will-change: transform;"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M3.7 6.7l3.8-3.8V15h1V2.9l3.8 3.8.7-.7-5-5-5 5z"
                    />
                  </svg>
                </button>
                <button
                  class="b-task-text-area__button bx--btn bx--btn--sm bx--btn--ghost bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--bottom bx--tooltip--align-center"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="bx--assistive-text"
                  >
                    Find next
                  </span>
                  <svg
                    aria-hidden="true"
                    aria-label="Find next"
                    class="bx--btn__icon"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    style="will-change: transform;"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12.3 9.3l-3.8 3.8V1h-1v12.1L3.7 9.3 3 10l5 5 5-5z"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="react-codemirror2"
              >
                <div
                  class="CodeMirror cm-s-material CodeMirror-wrap"
                >
                  <div
                    style="overflow: hidden; position: relative; width: 3px; height: 0px;"
                  >
                    <textarea
                      autocapitalize="off"
                      autocorrect="off"
                      spellcheck="false"
                      style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"
                      tabindex="0"
                    />
                  </div>
                  <div
                    class="CodeMirror-vscrollbar"
                    cm-not-content="true"
                    tabindex="-1"
                  >
                    <div
                      style="min-width: 1px;"
                    />
                  </div>
                  <div
                    class="CodeMirror-hscrollbar"
                    cm-not-content="true"
                    tabindex="-1"
                  >
                    <div
                      style="height: 100%; min-height: 1px;"
                    />
                  </div>
                  <div
                    class="CodeMirror-scrollbar-filler"
                    cm-not-content="true"
                  />
                  <div
                    class="CodeMirror-gutter-filler"
                    cm-not-content="true"
                  />
                  <div
                    class="CodeMirror-scroll"
                    tabindex="-1"
                  >
                    <div
                      class="CodeMirror-sizer"
                      style="margin-left: 0px;"
                    >
                      <div
                        style="position: relative;"
                      >
                        <div
                          class="CodeMirror-lines"
                          role="presentation"
                        >
                          <div
                            role="presentation"
                            style="position: relative; outline: none;"
                          >
                            <div
                              class="CodeMirror-measure"
                            >
                              <pre
                                class="CodeMirror-line-like"
                              >
                                <span>
                                  xxxxxxxxxx
                                </span>
                              </pre>
                            </div>
                            <div
                              class="CodeMirror-measure"
                            />
                            <div
                              style="position: relative; z-index: 1;"
                            />
                            <div
                              class="CodeMirror-cursors"
                            />
                            <div
                              class="CodeMirror-code"
                              role="presentation"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      style="position: absolute; height: 50px; width: 1px;"
                    />
                    <div
                      class="CodeMirror-gutters"
                      style="left: 0px;"
                    >
                      <div
                        class="CodeMirror-gutter CodeMirrorReact-linenumbers"
                      />
                      <div
                        class="CodeMirror-gutter CodeMirror-foldgutter"
                      />
                      <div
                        class="CodeMirror-gutter CodeMirror-linenumbers"
                        style="width: 1px;"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
      </section>
    </form>
  </div>
  <div
    class="ReactModalPortal"
  />
</body>
`;
